treeherder/push_health/similar_jobs.py

```diff
@@ -0,0 +1,63 @@
+from itertools import groupby
+
+from django.db.models import Q
+from django.forms.models import model_to_dict
+
+from treeherder.model.models import Job
+
+job_fields = [
+    'id',
+    'machine_platform',
+    'option_collection_hash',
+    'job_type',
+    'result',
+    'failure_classification'
+]
+
+
+def set_matching_passed_jobs(failures, push):
+
+    failed_jobs = {}
+    for failure in failures:
+        jobs = {job['id']: job for job in failure['failJobs']}
+        failed_jobs.update(jobs)
+
+    # Need to OR for each type of job that failed
+    query_conditions = []
+    for job in failed_jobs.values():
+        query_conditions.append(Q(
+            option_collection_hash=job['option_collection_hash'],
+            machine_platform=job['machine_platform'],
+            job_type=job['job_type'],
+        ))
+    query = query_conditions.pop()
+    for condition in query_conditions:
+        query |= condition
```

Comment:
what are we expecting the value of query to be?

Reply:
``query`` will end up being a bunch of OR conditions for each type of job we want to find that's in a "success" state.  For instance: ``(platform=x, config=y, job_type=z OR platform=a, config=b, job_type=c OR ...)